crates/sui-cost-tables/src/bytecode_tables.rs

```diff
@@ -47,6 +50,9 @@ static ZERO_COST_SCHEDULE: Lazy<CostTable> = Lazy::new(zero_cost_schedule);
 pub struct GasStatus<'a> {
     cost_table: &'a CostTable,
     gas_left: InternalGas,
+    vm_gas_left: InternalGas,
+    /// This is used for batches where we want to ensure we don't reserve multiple times
+    reserved: bool,
```

Comment:
I'm a bit confused by this. Shouldn't we reserve the flat fee for each VM execution in the batch?

Reply:
See other comment. I'm open to either idea, but I simply picked the conservative one for now.

Reply:
logic has been simplified